src/amo/components/OverallRating.js

```diff
@@ -1,37 +1,69 @@
 import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
 
+import { setUserRating } from 'amo/actions/ratingActions';
+import { callApi } from 'core/api';
 import translate from 'core/i18n/translate';
+import log from 'core/logger';
 
 import 'amo/css/OverallRating.scss';
 
 
 export class OverallRatingBase extends React.Component {
   static propTypes = {
-    i18n: PropTypes.object,
-    addonName: PropTypes.string,
+    addonName: PropTypes.string.isRequired,
+    addonID: PropTypes.number.isRequired,
+    authToken: PropTypes.string,
+    version: PropTypes.object.isRequired,
+    i18n: PropTypes.object.isRequired,
+    createRating: PropTypes.func.isRequired,
+  }
+
+  onChange = ({ currentTarget: form }) => {
+    const input = form.querySelector('input:checked');
+    if (!input) {
+      // I don't think any user input would cause a form change like
+      // this. I'm not sure what would cause this at all but it's probably
+      // better not to throw an exception.
+      log.error('The form changed but no input was selected');
```

Comment:
This does seem strange, it feels like radio buttons might not be the best here. I wouldn't expect clicking a radio button to submit a form necessarily but I could imagine there being an auto-save on change so I guess that's kind of what this is doing.

It does seem like switching to a `<button>` would simplify this though. It could submit the form or use `onClick` and then there isn't really any ambiguity. In the future I think this will bring up the "add what you like/don't like" page so I feel a button makes more sense for that, too.